File: Week7_1/1.1/1.1/Square.m



//
//  Square.m
//  1.1
//
//  Created by kongyunpeng on 4/16/16.
//  Copyright © 2016 universal. All rights reserved.
//

#import "Square.h"

@implementation Square
- (instancetype)initWithSide:(double)side
{
    self = [super init];
    if (self) {
        self.side = side;
    }
    return self;
}
-(double)perimeter;{
    return  _side * 4;
}
-(double)area {
    return _side * _side;
}


@end

File: Week7_1/1.2/1.2/Square.swift

//
//  Square.swift
//  1.2
//
//  Created by kongyunpeng on 4/16/16.
//  Copyright © 2016 universal. All rights reserved.
//

import Foundation
class Square {
    var side:Double
    init(side:Double) {
        self.side = side
    }
    
    func perimeter() -> Double{
        return 4 * self.side
    }
    
    func area() -> Double {
        return side * side
    }
}